frame/bounties/src/lib.rs

```diff
@@ -660,12 +789,685 @@ decl_module! {
 					},
 					_ => return Err(Error::<T>::UnexpectedStatus.into()),
 				}
-
 				Ok(())
 			})?;
 
 			Self::deposit_event(Event::<T>::BountyExtended(bounty_id));
 		}
+
+		/// Add a new subbounty.
+		///
+		/// The dispatch origin for this call must be master curator.
+		/// parent bounty must be in "active" state.
+		///
+		/// Subbouty gets added successfully & fund gets transferred from
+		/// parent bounty to subbounty account, if parent bounty has
+		/// enough fund. Else call gets failed.
+		///
+		/// Upper bound to maximum number of active  subbounties that
+		/// can be added are managed via runtime trait config
+		/// 'MaxActiveSubBountyCount'.
+		///
+		/// If the call is success, the state of subbounty is
+		/// moved to "Added" state.
+		///
+		/// - `bounty_id`: Bounty ID for which subbounty to be added.
+		/// - `value`: Value for executing the proposal.
+		/// - `description`: Text description for the subbounty.
+		#[weight = <T as Config>::WeightInfo::add_subbounty(description.len() as u32)]
+		fn add_subbounty(
+			origin,
+			#[compact] bounty_id: BountyIndex,
+			value: BalanceOf<T>,
+			description: Vec<u8>,
+		) {
+			let signer = ensure_signed(origin)?;
+
+			Bounties::<T>::try_mutate_exists(
+				bounty_id,
+				|maybe_bounty| -> DispatchResult {
+					let bounty = maybe_bounty
+						.as_mut()
+						.ok_or(Error::<T>::InvalidIndex)?;
+
+					if let BountyStatus::Active { ref curator, .. } = bounty.status {
+						ensure!(signer == *curator, Error::<T>::RequireCurator);
+
+						// Verify the arguments
+						ensure!(
+							description.len() <= T::MaximumReasonLength::get() as usize,
+							Error::<T>::ReasonTooBig,
+						);
+						ensure!(
+							value >= T::BountyValueMinimum::get(),
+							Error::<T>::InvalidValue,
+						);
+						ensure!(
+							bounty.active_subbounty_count <
+								T::MaxActiveSubBountyCount::get() as u32,
+							Error::<T>::TooManySubBounties,
+						);
+
+						// Read parent bounty account info
+						let bounty_account = Self::bounty_account_id(bounty_id);
+
+						// Create subbounty ID.
+						let subbounty_id = Self::bounty_count();
+
+						// Transfer fund from parent bounty to subbounty.
+						let subbounty_account = Self::bounty_account_id(subbounty_id);
+						T::Currency::transfer(
```

Comment:
what if `bounty_account ` has exactly 0 left after this call?

Reply:
Extrinsic call `add_subbounty()` passes, without any issue.

I hope your question is about, consequence of bounty account if balance drops below `ExistentialDeposit`.

As part of `post_mutation()` in pallet-balance, `account` may be removed from storage instance.

Looks like corner case, Just wondering, can we fail the transaction with error of `InsufficientBountyBalance` ?

Reply:
Yeah I think so

Reply:
Done :+1: